src/seedu/addressbook/data/person/Address.java

```diff
@@ -36,24 +33,12 @@ public static boolean isValidAddress(String test) {
         return test.matches(ADDRESS_VALIDATION_REGEX);
     }
 
-    @Override
-    public String toString() {
-        return value;
-    }
 
     @Override
     public boolean equals(Object other) {
         return other == this // short circuit if same object
                 || (other instanceof Address // instanceof handles nulls
-                && this.value.equals(((Address) other).value)); // state check
+                && super.value.equals(((Address) other).value)); // state check
```

Comment:
you can still use this.value here, as members of the parent class are inherited by the child class.